### PR TITLE
Support codecov (with configuration) for test_with_coverage

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: mono_repo/lib/src/basic_config.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'coverage_processor.dart';
+
+abstract class BasicConfiguration {
+  Set<CoverageProcessor> get coverageProcessors;
```
Comment (kevmoo) [inline mono_repo/lib/src/basic_config.dart:8]: Trying to minimize the amount of random state we sent to individual tasks. Not strictly required. Happy to discuss.

File: mono_repo/lib/src/mono_config.dart [hunk 2]
```diff
@@ -27,41 +30,25 @@ const _allowedPubActions = {
...
 
-class MonoConfig {
+class MonoConfig implements BasicConfiguration {
   final Map<String, ConditionalStage> githubConditionalStages;
   final Set<String> mergeStages;
   final bool prettyAnsi;
   final String pubAction;
   final String? selfValidateStage;
   final GitHubConfig github;
-
```
Comment (kevmoo) [inline mono_repo/lib/src/mono_config.dart:37]: Unrelated cleanup 🤷

File: mono_repo/lib/src/basic_config.dart [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'coverage_processor.dart';
+
+abstract class BasicConfiguration {
```
Comment (jakemac53) [inline mono_repo/lib/src/basic_config.dart:7]: nit: name should be more descriptive, BasicCoverageConfiguration etc. Or if its intended for more than that in the future then give it comment or something describing the intended use

**Review by jakemac53 — APPROVED** [review]

File: mono_repo/lib/src/basic_config.dart (hunk 3 above)
Comment (kevmoo) [reply mono_repo/lib/src/basic_config.dart:11]: Done!
